src/lib/WalletInfoCollector.ts

```diff
@@ -82,10 +89,24 @@ export default class WalletInfoCollector {
         }
         onUpdate(walletInfo, await derivedAccountsPromise);
 
-        if (walletType === WalletType.LEGACY) {
+        if (walletType === WalletType.LEGACY && WalletInfoCollector._keyguardClient) {
             // legacy wallets have no derived accounts
             await initialAccountsPromise;
-            return { walletInfo };
+            const address = initialAccounts[0].address;
+
+            if (checkLegacyActivity) {
+                hasHistoryOrBalance = walletInfo.accounts.get(address)!.balance !== undefined &&
```

Comment:
have to use
```ts
hasActivity = hasActivity || ....
```
here to avoid overwriting the information whether the address owns a contract.